client/src/components/CreatePost.tsx

```diff
@@ -0,0 +1,225 @@
+import React, { useState } from "react";
+import CssBaseline from "@mui/material/CssBaseline";
+
+import { createTheme, ThemeProvider } from "@mui/material/styles";
+import Grid from "@mui/material/Grid";
+import Box from "@mui/material/Box";
+import Container from "@mui/material/Container";
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Divider from "@mui/material/Divider";
+import Typography from "@mui/material/Typography";
+
+import Dialog from "@mui/material/Dialog";
+import DialogContent from "@mui/material/DialogContent";
+import DialogActions from "@mui/material/DialogActions";
+
+import PreviewPopUp from "./PreviewPopUp";
+import { DialogTitle } from "@mui/material";
+
+const theme = createTheme();
+
+function CreatePost() {
+  // create hooks for input fields
+  const [title, setTitle] = useState("");
+  const [body, setBody] = useState("");
+  const [file, setFile] = useState("");
+  const [tags, setTags] = useState("");
+  const [eventCapacity, setEventCapacity] = useState("");
+  const [eventLocation, setEventLocation] = useState("");
+
+  const [openPopup, setOpenPopup] = useState(false);
+  const [openPopupError, setOpenPopupError] = useState(false);
+
+  const allFields = [title, body, file];
+
+  // handle functions
+  const handleClickOpen = () => {
+    let noEmptyFields = true;
+    allFields.forEach((inputField) => {
+      if (inputField === "") {
+        noEmptyFields = false;
+      }
+    });
+    if (noEmptyFields) {
+      setOpenPopup(true);
+    } else {
+      setOpenPopupError(true);
+    }
+    return;
```

Comment:
```suggestion
    if (allFields.map(e => !e.trim()).some(e => e)) { // if any input field is empty
          setOpenPopupError(true);
    } else {
      setOpenPopup(true);
    }
```

`allFields.map(e => !e.trim()).some(e => e)` will return `true` if one or more strings are empty. Has [good support](https://caniuse.com/mdn-javascript_builtins_array_some) as well